db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/BinaryListOperatorNode.java

```diff
@@ -252,8 +266,13 @@ public void bindComparisonOperator() throws StandardException{
 
         /* Can the types be compared to each other? */
         /* Multicolumn IN list cannot currently be constructed before bind time. */
-        if (singleLeftOperand)
-            rightOperandList.comparable(getLeftOperand());
+        if (singleLeftOperand) {
+            if (!rightOperandList.comparable(getLeftOperand(), false)) {
```

Comment:
`ValueNodeList.comparable()` is only used here, I'd rather make it always return a boolean and explicitly raise an exception here (instead of adding the extra boolean parameter), to me it's surprising that comparable() would raise an exception instead of returning true/false